browser-extension/mv3/src/service-worker/services/clientHandler.ts

```diff
@@ -1,12 +1,18 @@
-import { generateUrlPattern } from "../../utils";
+import { generateUrlPattern, isExtensionEnabled } from "../../utils";
 import { WEB_URL, OTHER_WEB_URLS } from "../../../../config/dist/config.build.json";
-import { isExtensionEnabled } from "../../utils";
 import { Variable, onVariableChange } from "../variable";
 import { RuleType } from "common/types";
 import rulesStorageService from "../../rulesStorageService";
+import { getBlockedDomains } from "./utils";
+import { ChangeType, onRecordChange } from "common/storage";
+import { STORAGE_KEYS } from "common/constants";
 
 const excludeMatchesPatterns = [WEB_URL, ...OTHER_WEB_URLS].map(generateUrlPattern).filter((pattern) => !!pattern);
 
+const generateBlockedDomainPattern = (hostname: string) => {
+  return `*://*${hostname}/*`;
+};
+
```

Comment:
- hostname -> host (as it can contain port too)
- Rename to `generateBlockedHostMatchPattern`
- This will match domains which shouldn't (`example.com` then it would match `theexample.com` too. We only need to match current domain and all subdomains)